File: WinPaletter/Classes/INI.vb
Imports System.Text
Imports WinPaletter.NativeMethods.Kernel32

Public Class INI : Implements IDisposable

    Public path As String
    Private disposedValue As Boolean

    Public Sub New(ByVal INIPath As String)
        path = INIPath
    End Sub

    Public Sub IniWriteValue(ByVal Section As String, ByVal Key As String, ByVal Value As String)
        WritePrivateProfileString(Section, Key, Value, Me.path)
    End Sub

    Public Function IniReadValue(ByVal Section As String, ByVal Key As String, Optional DefaultValue As String = Nothing) As String
        Dim temp As StringBuilder = New StringBuilder(65535)
        Dim i As Integer = GetPrivateProfileString(Section, Key, DefaultValue, temp, temp.Capacity, Me.path)
        Return temp.ToString()
    End Function

    Protected Overridable Sub Dispose(disposing As Boolean)
        If Not disposedValue Then
            If disposing Then
                ' TODO: dispose managed state (managed objects)
            End If

            ' TODO: free unmanaged resources (unmanaged objects) and override finalizer
            ' TODO: set large fields to null
            disposedValue = True
        End If
    End Sub

    ' ' TODO: override finalizer only if 'Dispose(disposing As Boolean)' has code to free unmanaged resources
    ' Protected Overrides Sub Finalize()
    '     ' Do not change this code. Put cleanup code in 'Dispose(disposing As Boolean)' method
    '     Dispose(disposing:=False)
    '     MyBase.Finalize()
    ' End Sub

    Public Sub Dispose() Implements IDisposable.Dispose
        ' Do not change this code. Put cleanup code in 'Dispose(disposing As Boolean)' method
        Dispose(disposing:=True)
        GC.SuppressFinalize(Me)
    End Sub
End Class

File: WinPaletter/Classes/INI.cs
using System;
using System.Text;
using WinPaletter.NativeMethods;

namespace WinPaletter
{

    public class INI : ICloneable, IDisposable

    {
        public string path;
        private bool disposedValue;

        public INI(string File)
        {
            path = File;
        }

        public void Write(string Section, string Key, string Value)
        {
            Kernel32.WritePrivateProfileString(Section, Key, Value, path);
        }

        public string Read(string Section, string Key, string DefaultValue = null)
        {
            StringBuilder SB = new(65535);
            int i = Kernel32.GetPrivateProfileString(Section, Key, DefaultValue, SB, SB.Capacity, path);
            return SB.ToString();
        }

        public void DeleteSection(string Section)
        {
            Kernel32.WritePrivateProfileString(Section, null, null, path);
        }

        public void DeleteKey(string Section, string Key)
        {
            Kernel32.WritePrivateProfileString(Section, Key, null, path);
        }

        #region Clone support
        public object Clone()
        {
            return MemberwiseClone();
        }
        #endregion

        #region IDisposable Support
        protected virtual void Dispose(bool disposing)
        {
            if (!disposedValue)
            {
                if (disposing)
                {
                    // TODO: dispose managed state (managed objects)
                }

                // TODO: free unmanaged resources (unmanaged objects) and override finalizer
                // TODO: set large fields to null
                disposedValue = true;
            }
        }

        // // TODO: override finalizer only if 'Dispose(bool disposing)' has code to free unmanaged resources
        // ~INI()
        // {
        //     // Do not change this code. Put cleanup code in 'Dispose(bool disposing)' method
        //     Dispose(disposing: false);
        // }

        public void Dispose()
        {
            // Do not change this code. Put cleanup code in 'Dispose(bool disposing)' method
            Dispose(disposing: true);
            GC.SuppressFinalize(this);
        }

        #endregion
    }
}